### PR TITLE
Fix cases when JsonMasker can get in infinite loop

### PR DESCRIPTION
**Pull request by gavlyukovskiy**
Fixes issue when we cannot find the end of an array, other validation is mostly for failing fast with more predictable error message.
Added a formatter to produce an invalid json.

### CODE REVIEW

File: src/test/java/dev/blaauwendraad/masker/json/InvalidJsonTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class InvalidJsonTest {
```
Comment (gavlyukovskiy) [inline src/test/java/dev/blaauwendraad/masker/json/InvalidJsonTest.java:15]:
```suggestion
class InvalidJsonTest {
```

**Review by donavdey — APPROVED** [review]
I was not able to come up with examples that would stuck the masker. So I'd say it is safe enough.

File: src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java [hunk 2]
```diff
@@ -178,15 +190,15 @@ private void visitObject(MaskingState maskingState, @CheckForNull KeyMaskingConf
...
+            // check if we're at the end of a (non-empty) object
+            if (maskingState.byteAtCurrentIndex() == '}') {
+                break;
             }
-            stepOverWhitespaceCharacters(maskingState);
-
-            maskingState.backtrackCurrentJsonPath();
         }
-        maskingState.incrementCurrentIndex(); // step over closing curly bracket
+        maskingState.next();
```
Comment (Breus) [inline src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java:201]: Nit: it is nice to add the comment back (step over closing curly bracket) but if this is the only comment I can do it in my next PR

File: src/test/java/dev/blaauwendraad/masker/randomgen/InvalidJsonPrettyPrinter.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            '[',
+            ']',
+            ':',
+            '\'',
+            '"',
+            ',',
+            'a',
+            't',
+            'f',
+            'n'
```
Comment (Breus) [inline src/test/java/dev/blaauwendraad/masker/randomgen/InvalidJsonPrettyPrinter.java:34]: Probably we should also add numeric characters (-, e, E, 0, 1) but can also do it in later PR

**Review by Breus — APPROVED** [review]

**Pull request merged**